### ISSUE TITLE
Adding PSI_API prefix for AOIntegralsIterator 

### ISSUE THREAD

**Issue by Masterluke87** [psi4/psi4#2239]
Hi,

we were recently reviving a PSI4 plugin that was used with an older version of PSI4. Now, we wanted to adapt it to the current development version and realized that the AOIntegralsIterator is hidden (?) in the library. Our fix was to add PSI_API to the class definition in `src/psi4/libmints/integral.h` . 

However, for us, it would be easier if it would be accessible by default (mainly to use the conda install and avoid compilation every time). Is there a specific reason, why that class is excluded or hidden in the library? If no, we would be very grateful, if a developer would add it to the header file. 

Thank you!

**Comment by JonathonMisiewicz** [comment]
Classes are hidden by default because when we expose a class as `PSI_API`, changing that class affects plugin developers, so we need to add warnings before we actually make the changes. And that takes time. So we can expose a class if it's useful, but we're not going to expose a class by default.

Could you elaborate on what your usecase is for this particular class? Before exposing the class, I'd like to make sure you really do need this class instead of something else in the `libmints` machinery. I believe that some pieces moved around during our recent upgrade to libint2.

**Comment by jturney** [comment]
Yeah, that code isn't efficient to use. That's probably why it wasn't exported. Could we help you to update the code instead?

**Comment by Masterluke87** [comment]
Many thanks for your answers and support. We basically want to export the two-electron integrals, and use them with other programs. So the integrals will be written to a binary file together with other 1el matrices.  The code snippet that is working at the moment:

``` c++
MintsHelper mints(ref_wfn->basisset(), options, 0);
auto intFactory = mints.integral();
auto basis         = mints.basisset();
auto tBO           = intFactory->eri();
AOShellCombinationsIterator shellIter = intFactory->shells_iterator();

std::vector<double> intval;
std::vector<int>  intnums;

for (shellIter.first(); shellIter.is_done() == false; shellIter.next()) {
    tBO->compute_shell(shellIter.p(),shellIter.q(),shellIter.r(),shellIter.s());
    double cutOff = 1E-12;
    auto aoIter = intFactory->integrals_iterator(shellIter.p(),shellIter.q(),shellIter.r(),shellIter.s());
    for (aoIter.first(); aoIter.is_done() == false; aoIter.next()) {
            int i = aoIter.i();
            int j = aoIter.j();
            int k = aoIter.k();
            int l = aoIter.l();

            if (tBO->buffer()[aoIter.index()]>cutOff){
                    intval.push_back(tBO->buffer()[aoIter.index()]);
                    intnums.push_back(i);                        
                    intnums.push_back(j);
                    intnums.push_back(k);
                    intnums.push_back(l);

            }

    }
   // Continue Sorting + Exporting
````

If we use the conda install, we get an error that the symbol cannot be found. Compiling with modified integral.h works as expected.


Thanks again!
